Security/src/AuthClient/Program.cs

```diff
@@ -0,0 +1,107 @@
+using Microsoft.AspNetCore.Authentication.Cookies;
+using Microsoft.AspNetCore.Authentication.OpenIdConnect;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.HttpOverrides;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Steeltoe.Common;
+using Steeltoe.Common.Certificates;
+using Steeltoe.Configuration.CloudFoundry;
+using Steeltoe.Configuration.CloudFoundry.ServiceBinding;
+using Steeltoe.Management.Endpoint;
+using Steeltoe.Samples.AuthClient;
+using Steeltoe.Security.Authentication.OpenIdConnect;
+using Steeltoe.Security.Authorization.Certificate;
+using System;
+using System.Linq;
+using System.Net.Http;
+
+const string organizationId = "a8fef16f-94c0-49e3-aa0b-ced7c3da6229";
+const string spaceId = "122b942a-d7b9-4839-b26e-836654b9785f";
+
+WebApplicationBuilder builder = WebApplication.CreateBuilder(args);
+
+builder.Configuration
+    .AddCloudFoundry() // needed for actuators
+    .AddCloudFoundryServiceBindings()
+    .AddAppInstanceIdentityCertificate(new Guid(organizationId), new Guid(spaceId));
+
+builder.Services
+    .AddAuthentication(options =>
+    {
+        options.DefaultScheme = CookieAuthenticationDefaults.AuthenticationScheme;
+        options.DefaultChallengeScheme = OpenIdConnectDefaults.AuthenticationScheme;
+    })
+    .AddCookie(options =>
+    {
+        options.AccessDeniedPath = new PathString("/Home/AccessDenied");
+    })
+    .AddOpenIdConnect()
+    .ConfigureOpenIdConnectForCloudFoundry();
+
+builder.Services.AddSession();
+builder.Services.AddAuthorizationBuilder()
+    .AddPolicy(Globals.RequiredJwtScope, policy => policy.RequireClaim("scope", Globals.RequiredJwtScope))
+    .AddPolicy(Globals.UnknownJwtScope, policy => policy.RequireClaim("scope", Globals.UnknownJwtScope));
+
+if (Platform.IsCloudFoundry)
+{
+    builder.Services.RegisterCloudFoundryApplicationInstanceInfo();
+}
+else
+{
+    builder.Services.RegisterDefaultApplicationInstanceInfo();
+}
```

Comment:
Doesn't Steeltoe offer an over-arching method, so that users don't need to care about the specific platform?

Reply:
Steeltoe should, but I do not believe there is one yet. This is something we should consider while reviewing Steeltoe.Common